### ISSUE TITLE
Questions Regarding PAGE_SIZE and SQLITE_ESP32VFS_BUFFERSZ

### ISSUE THREAD

**Issue by slowpoison4** [siara-cc/esp32_arduino_sqlite3_lib#86]
Hi 
This is more of a Question/Observation. I have a very large program on ESP32 which uses the SQlite3 Library. Very early on i ran into the dreaded Sql %s error "out of memory". I dug through the code (Im a novice so not really sure about a lot of the implementation), But i saw  SQLITE_ESP32VFS_BUFFERSZ  defined as 8196. I decided to reduce this to 64 (Aggressive) but i saw a marked improvement in RAM usage. I wasnt getting the error any more and my code works fine. 

For reference, My code structure has 2 tables with 5 columns each storing texts. each column field size is about 192 characters. 

I also reduced the page size to 64 and once again saw major improvement in RAM utilization. I never faced issues with adding data, but deleting always failed before this. Now i can store 250+ messages on LittleFS with virtually no Performance impact. 

Am i doing something wrong here? I dont get why these changes worked because im dont know the importance of these values. Could you please shed some light on the same?

Once again, Big FAN of this library and really appreciate the work gone into this

This is my Config_ext.h
#define BUILD_sqlite -DNDEBUG
#define SQLITE_OMIT_LOAD_EXTENSION 1
#define SQLITE_DQS 0
#define SQLITE_OS_OTHER 1
#define SQLITE_NO_SYNC 1
#define SQLITE_TEMP_STORE 1
#define SQLITE_DISABLE_LFS 1
#define SQLITE_DISABLE_DIRSYNC 1
#define SQLITE_SECURE_DELETE 0
#define SQLITE_DEFAULT_LOOKASIDE 512, 64
#define YYSTACKDEPTH 20
#define SQLITE_SMALL_STACK 1
#define SQLITE_DEFAULT_PAGE_SIZE 64
#define SQLITE_SORTER_PMASZ 4
#define SQLITE_DEFAULT_CACHE_SIZE -1
#define SQLITE_DEFAULT_MEMSTATUS 0
#define SQLITE_DEFAULT_MMAP_SIZE 0
#define SQLITE_CORE 1
#define SQLITE_SYSTEM_MALLOC 1
#define SQLITE_THREADSAFE 1
#define SQLITE_MUTEX_APPDEF 1
#define SQLITE_OMIT_WAL 1
#define SQLITE_DISABLE_FTS3_UNICODE 1
#define SQLITE_DISABLE_FTS4_DEFERRED 1
#define SQLITE_LIKE_DOESNT_MATCH_BLOBS 1
#define SQLITE_DEFAULT_FOREIGN_KEYS 0
#define SQLITE_DEFAULT_LOCKING_MODE 1
#define SQLITE_DEFAULT_PAGE_SIZE 64
#define SQLITE_DEFAULT_PCACHE_INITSZ 8
#define SQLITE_MAX_DEFAULT_PAGE_SIZE 32768
#define SQLITE_POWERSAFE_OVERWRITE 1
#define SQLITE_MAX_EXPR_DEPTH 0

**Comment by siara-cc** [comment]
Hi @slowpoison4 
Thanks for the feedback!  About your queries:
- SQLITE_ESP32VFS_BUFFERSZ is just a buffer before making a write to disk so yes reducing it could make it occupy lesser memory.  I thought 8192 is not too much for esp32 but looks like it is worthwhile to reduce it.
- The minimum value for SQLITE_DEFAULT_PAGE_SIZE is 512 and not sure how it works for 64. It probably uses the higher value of sector size and this value.  I reduced the default page size to 512 to improve memory consumption. Probably you have a copy that is before this change.
Thanks for sharing your experience!

**Comment by slowpoison4** [comment]
Hi @siara-cc 

Thanks for the quick response. I tried the Default Page size with the latest release as well, and I observed the same effect as before. The stability is fine as long as i use a Mutex to ensure no other task accesses/opens the db before the previous request is complete. I cannot open the db on my laptop with dB browser. It always gives me a corrupt file error (I suspect the page size being the cause here). But since i dont intend to use it in my laptop and purely using it for offline storage and retrieval it works perfect for me. 

Thanks for your time !
